tests: update encrypted partition creation test

cmatsuoka opened the pull request:
Update the create-partitions-encrypted test to handle sealed keys. This
test requires a spread back-end that supports TPM and secure boot (thanks
Sergio!).

Note: this also requires a spread binary that supports systems with the
secureboot attribute (https://github.com/snapcore/spread/pull/104).

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

sergiocazzolato commented:
Hi, the test is ok, it is missing now to run it. I think the easyest way to do it is adding a new system in google backend and removing the google-tpm backend.

this should be the new system under google backend

            - ubuntu-20.04-64-secboot:
                workers: 1
                secureboot: true


this should be the system lists in the github workflow
      matrix:
        system:
        - ubuntu-19.10-64
        - ubuntu-20.04-64
        - ubuntu-20.04-64-secboot
        - ubuntu-core-20-64
        - debian-sid-64
        - fedora-30-64
        - fedora-31-64

Then the tests should be updated to filter accordingly

cmatsuoka commented:
Thanks for reviewing. I removed the google-tpm backend, added the ubuntu-20.04-64-secboot system as specified under the google backend, and updated the systems list in encryption test and the github workflow matrix as directed.

cmatsuoka commented:
Changes in this PR were merged into PR #8551.
